tools/nightly_build_and_test_stats.groovy

```diff
@@ -325,6 +359,86 @@ def verifyReleaseContent(String version, String release, String variant, Map sta
     }
 }
 
+// For a given pipeline, tell us how reproducible the builds were.
+// Note: Will limit itself to jdk versions and platforms in the results Map.
+def getReproducibilityPercentage(String jdkVersion, String trssId, String trssURL, Map results) {
+    echo "Called repro method with trssID:"+trssId
+
+    def platformConversionMap = getPlatformConversionMap()
+    def platformReproTestMap = getPlatformReproTestMap()
+
+    // We are only looking for reproducible percentages for the relevant jdk versions...
+    if ( trssId != "" && results.containsKey(jdkVersion) ) {
+        def jdkVersionInt = jdkVersion.replaceAll("[a-z]", "")
+
+        // ...and platforms.
+        results[jdkVersion][1].each { onePlatform, valueNotUsed ->
+            // If this platform doesn't have a reproducibility test yet, skip it.
+            if (platformReproTestMap[onePlatform][0].equals("NA")) {
+                results[jdkVersion][1][onePlatform] = "NA"
+                // Then we exit this lambda and skip to the next platform.
+                return
+            }
+
+            def pipelineLink = trssURL+"/api/getAllChildBuilds?parentId="+trssId+"\\&buildNameRegex=^"+jdkVersion+"\\-"+platformConversionMap[onePlatform][0]+"\\-temurin\$"
+            def trssBuildJobNames = sh(returnStdout: true, script: "wget -q -O - ${pipelineLink}")
+            def platformResult = "???% - Build not found. Pipeline link: " + pipelineLink
+
+            // Does this platform have a build in this pipeline?
+            if ( trssBuildJobNames.length() > 10 ) {
+                def buildJobNamesJson = new JsonSlurper().parseText(trssBuildJobNames)
+
+                // For each build, search the test output for the unit test we need, then look for reproducibility percentage.
+                buildJobNamesJson.each { buildJob ->
+                    platformResult = "???% - Build found, but no reproducibility tests. Build link: " + buildJob.buildUrl
+                    def testPlatform = platformConversionMap[onePlatform][1]
+                    def reproTestName=platformReproTestMap[onePlatform][1]
+                    def reproTestBucket=platformReproTestMap[onePlatform][0]
+                    def testJobTitle="Test_openjdk${jdkVersionInt}_hs_${reproTestBucket}_${testPlatform}.*"
+                    def trssTestJobNames = sh(returnStdout: true, script: "wget -q -O - ${trssURL}/api/getAllChildBuilds?parentId=${buildJob._id}\\&buildNameRegex=^${testJobTitle}\$")
+
+                    // Did this build have tests?
+                    if ( trssTestJobNames.length() > 10 ) {
```

Comment:
ditto "10", if significant maybe add reason in comment?

Reply:
As above. :)